### PR TITLE
[WIP] Frame view for HFB data

### PR DESCRIPTION
**Pull request by james-s-willis**
Could you review my `hfbFrameView` and `hfbFileRaw`? @andrerenard @jrs65 

I copied `visFrameView` and removed fields not relevant to HFB data. I did the same with `visFileRaw`.

### CODE REVIEW

**Comment by jrs65** [comment]
Thanks @james-s-willis I should be able to have a look over it later today.

**Comment by james-s-willis** [comment]
Can you review this, @jrs65 @andrerenard? So I can go ahead and create an abstract `frameView` class. I just need to identify the commonality between the `vis` and `hfb` frames. I also need to work out how it will fit in with the `visFile` class, I think we should have another call to discuss this.

File: lib/utils/hfbBuffer.hpp [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+
+    /// ID of the frequency bin
+    uint32_t freq_id;
+    /// ID of the dataset (vis, gatedvisX ...), main vis dataset = 0
+    uint64_t dataset_id;
+
+    /// Number of elements for data in buffer
+    uint32_t num_elements;
+    /// Number of products for data in buffer
+    uint32_t num_prod;
```
Comment (jrs65) [inline lib/utils/hfbBuffer.hpp]: I would suggest breaking this one down into `num_beams` and `num_subfreq` (or similar) parameters. I can see it being very useful having that information available even if you don't directly need it to decode the frame.

File: lib/utils/hfbBuffer.hpp [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+    /// The ctime of the integration frame
+    timespec ctime;
+    /// Nominal length of the frame in FPGA ticks
+    uint64_t fpga_seq_length;
+    // Amount of data that actually went into the frame (in FPGA ticks)
+    uint64_t fpga_seq_total;
+
+    /// ID of the frequency bin
+    uint32_t freq_id;
+    /// ID of the dataset (vis, gatedvisX ...), main vis dataset = 0
```
Comment (jrs65) [inline lib/utils/hfbBuffer.hpp]: I definitely think you should keep the dataset_id in here, but you probably want to update the comment which is fairly vis data specific.

File: lib/utils/hfbBuffer.cpp [hunk 3]
```diff
@@ -0,0 +1,248 @@
+#include "hfbBuffer.hpp"
+
+#include "gpsTime.h"
+
+#include "fmt.hpp"
+
+#include <set>
+
+
+template<typename T>
```
Comment (jrs65) [inline lib/utils/hfbBuffer.cpp]: These two are obviously shared with `visFrameView` and so we should move to a common location. It sounds like you're on top of that one though.

**Pull request closed without merging**